### PR TITLE
examples: use the official error_details.proto

### PR DESCRIPTION
**Pull request by dapengzhang0**
The `error_details.proto` file in `examples/src/main/proto/` was not originated from grpc-java. 
It was officially opensourced at https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto

### CODE REVIEW

File: examples/build.gradle [hunk 1]
```diff
@@ -25,6 +25,7 @@ repositories {
 def grpcVersion = '1.5.0-SNAPSHOT' // CURRENT_GRPC_VERSION
 
 dependencies {
+  compile "com.google.api.grpc:grpc-google-common-protos:0.1.12"
```
Comment (ericgribkoff) [inline examples/build.gradle]: I think this should use proto-google-common-protos, not grpc-google-common-protos. See https://github.com/grpc/grpc-java/issues/2961#issuecomment-298370880

Comment (dapengzhang0) [inline examples/build.gradle]: Thanks! Updated.

**Comment by ejona86** [comment]
Maven build is pretty broken. See the Travis logs.

**Comment by dapengzhang0** [comment]
That's interesting, the 1st commit using `grpc-google-common-protos` worked.

**Comment by ejona86** [comment]
FYI: this will conflict with #3150

**Comment by dapengzhang0** [comment]
Is there any particular reason we use v0.1.9 in grpc-protobuf? Can we bump to v0.1.12 globally?

**Comment by dapengzhang0** [comment]
`proto-google-common-protos` v0.1.9 pom.xml
```
<groupId>com.google.protobuf</groupId>
<artifactId>protobuf-java</artifactId>
<version>3.2.0</version>
```
`proto-google-common-protos` v0.1.12 pom.xml
```
<groupId>com.google.protobuf</groupId>
<artifactId>protobuf-java</artifactId>
<version>3.3.0</version>
```

**Comment by ejona86** [comment]
Let's add maven-enforcer-plugin to the pom.xml to detect problems like this in the future. It's important for our users to do the same:

```xml
      <plugin>
        <groupId>org.apache.maven.plugins</groupId>
        <artifactId>maven-enforcer-plugin</artifactId>
        <version>1.4.1</version>
        <executions>
          <execution>
            <id>enforce</id>
            <goals>
              <goal>enforce</goal>
            </goals>
            <configuration>
              <rules>
                <requireUpperBoundDeps/>
              </rules>
            </configuration>
          </execution>
        </executions>
      </plugin>
```

**Comment by ejona86** [comment]
Actually, I'm just going to do the maven enforcer part in #3158.

**Pull request merged**